data/lib/src/network/config/endpoint.dart

```diff
@@ -51,6 +54,23 @@ extension ServicePathExtension on ServicePath {
     return '${Endpoint.rootPath}${path}/${parameter}';
   }
 
+  String generateEndpointPathWithParameterAndQuery({String parameter, List<String> queryParameters}) {
+    var completePath = '${Endpoint.rootPath}${path}';
+    if (parameter != null && parameter.isNotEmpty) {
+      completePath += '/${parameter}';
+    }
+    if (queryParameters != null && queryParameters.isNotEmpty) {
+      completePath += '?';
+      queryParameters.asMap().forEach((index, query) {
+        completePath += query;
+        if (index < queryParameters.length - 1) {
+          completePath += '&';
+        }
+      });
+    }
+    return completePath;
+  }
```

Comment:
@imGok @ngohado did you have any other idea

Reply:
In my PR I proposed a builder logic with 2 methods for query and path WDYT ?

Reply:
I just forgot the `&`

Reply:
```
  ServicePath withQueryParameters(List<String> queryParameters) {
    return ServicePath('${path}?${queryParameters.join("&")}');
  }
```

Reply:
Suggestion of @imGok  is good, but I prefer uses `Map` for `queryParameters` like:
```
  parameters.entries
      .map((e) => '${e.key}=${e.value}')
      .join('&');

Passing arguments like:
{
      'name': 'ngohado',
      'state': 'vietnam',
      ...
}
```

Reply:
Yup, I think `Map` is better.

Reply:
agree with map of @ngohado